services/agents/channel/channel_test.go

```diff
@@ -85,6 +90,8 @@ func setup(t *testing.T, connect func(*Channel) error, expected ...error) (agent
 	require.NoError(t, err, "failed to create stream")
 
 	teardown := func(t *testing.T) {
+		channelM.Lock()
```

Comment:
we can make test to not finish by using channels in `connect` function and in test